src/appleseed/foundation/math/correlatedmultijitter.h

```diff
@@ -0,0 +1,186 @@
+
+//
+// This source file is part of appleseed.
+// Visit http://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Kevin Masson, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef APPLESEED_FOUNDATION_MATH_CMJ_H
+#define APPLESEED_FOUNDATION_MATH_CMJ_H
+
+// appleseed.foundation headers.
+#include "foundation/math/vector.h"
+
+// Standard headers.
+#include <cassert>
+#include <cmath>
+
+namespace foundation
+{
+
+//
+// Correlated Multi-Jittered Sampling.
+//
+// References:
+//
+//   https://graphics.pixar.com/library/MultiJitteredSampling/paper.pdf
+//
+
+template <typename T>
+inline void cmj_compute_mn(T& m, T& n, const unsigned int N);
+
+template <typename T>
+inline Vector<T, 2> cmj_generate_sample(
+    unsigned int        s,
+    const unsigned int  m,
+    const unsigned int  n,
+    const unsigned int  N,
```

Comment:
I'm not too fond of having two parameters differing only by their case (`n` and `N`). Also it would be nice to add comments stating what each parameter is/does.